src/app/core/services/notifications/fcm-rest-notification.service.ts

```diff
@@ -233,19 +231,20 @@ export class FcmRestNotificationService extends FcmNotificationService {
         type: taskInfo.name,
         sourceType: DefaultSourcePrefix,
         appPackage: DefaultPackageName,
-        scheduledTime: new Date(notification.timestamp),
-      },
+        scheduledTime: new Date(notification.timestamp)
+      }
     }
   }
 
-  getNotificationEndpoint(projectId, subjectId) {
+  getNotificationEndpoint(projectId, subjectId, notificationId?) {
     return urljoin(
       this.appServerService.getAppServerURL(),
       this.PROJECT_PATH,
       projectId,
       this.SUBJECT_PATH,
       subjectId,
-      this.NOTIFICATIONS_PATH
+      this.NOTIFICATIONS_PATH,
+      notificationId ? notificationId.toString() : ''
```

Comment:
I would overload the function instead of using if/else
```ts
getNotificationEndpoint(projectId, subjectId, notificationId){
....
}
```
And
```ts
getNotificationEndpoint(projectId, subjectId) {
....
}
```